kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -1231,6 +1231,14 @@ object KyuubiConf {
       .checkValue(_ > 0, "must be positive number")
       .createWithDefault(Duration.ofMinutes(5).toMillis)
 
+  val KUBERNETES_SPARK_DELETE_DRIVER_POD_ON_TERMINATION_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.kubernetes.spark.deleteDriverPodOnTermination.enabled")
+      .doc("If set to true then Kyuubi server will delete the spark driver pod after " +
+        s"the application terminates for ${KUBERNETES_TERMINATED_APPLICATION_RETAIN_PERIOD.key}.")
+      .version("1.8.1")
```

Comment:
this feature may not mature for 1.8.1, after another thought:

- my previous suggestion is bad, which causes the configuration name misleading due to the Pod delete moment.
- users may want to only delete the Pod completed with exit code 0 but skip those abnormal exited ones, thus they can get more information to debug later.

Reply:
let me think again and send a follow-up PR tomorrow(or weekend)

Reply:
> users may want to only delete the Pod completed with exit code 0 but skip those abnormal exited ones, thus they can get more information to debug later.

Agree with this, maybe we can change this boolean config to enum policy config, such as NONE, ONLY_COMPLETED, ALL

Reply:
I see, we need also check whether the ApplicationInfo.name is null.

Reply:
for NOT_FOUND, it might cause NPE.

Reply:
for NPE, #5718